src/Form.js

```diff
@@ -68,11 +67,14 @@ export default class Form extends Component {
     }
   }
   onSubmit = (event) => {
-    const { onSubmit, onError } = this.props;
+    const { onSubmit, onError, reset } = this.props;
     const { formState } = this.state;
     event.preventDefault();
     if (formState.isValid()) {
       onSubmit(formState.getResource(), () => this.reset(this.props));
+      if (reset) {
```

Comment:
I think reset should be handled inside component will receive props.

Once you know the form is valid you should call Form again with reset set to true.

Imagine a case where you call `onSubmit`, call the back end, the call fails, you don't want to reset the form there.

So, react-formify does it inside component will receive props.

https://github.com/stashimi/react-formify/blob/d8f2b43cbead2f1e2182e8732e368b310a527696/src/Form.js#L45